content/blog/nuttx-motor-control-and-sensing/index.md

```diff
@@ -0,0 +1,512 @@
+---
+title: "NuttX for Motor Control and Sensing 1/2"
+date: 2025-04-28
+tags: ["NuttX", "Apache", "ESP32", "Motor", "IMU"]
+showAuthor: false
+authors:
+    - "filipe-cavalcanti"
+summary: "This article demonstrates how to implement motor control and speed sensing on an ESP32-C6 using NuttX RTOS. It covers setting up MCPWM for motor control, ADC for potentiometer reading, and quadrature encoder for speed measurement. The implementation showcases NuttX's real-time capabilities through a practical example using common peripherals and sensors."
+---
+
+## Introduction
+
+This series of articles demonstrates how to use multiple ESP32-C6 peripherals to build a complete motor control and sensing system. We'll explore:
+- Motor Control PWM (MCPWM) for DC motor speed control
+- Quadrature encoder (pulse counter) for motor speed measurement
+- ADC for real-time potentiometer position reading
+- Inertial Measurement Unit (IMU) via I2C for vibration analysis
+- Wireless connectivity for data acquisition and processing
+
+The goal is to build a comprehensive motor control and sensing system using the NuttX RTOS. We'll implement open-loop speed control while measuring mechanical vibrations through an IMU. The system will stream sensor data wirelessly to a computer for logging and analysis, showcasing NuttX's real-time capabilities, multi-threading support, and the ESP32-C6's wireless connectivity features.
```

Comment:
Remove last comma: `multi-threading support and the ESP32-C6's wireless connectivity features.`